pkg/volume/configmap/configmap.go

```diff
@@ -172,6 +172,10 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 
 	configMap, err := kubeClient.Core().ConfigMaps(b.pod.Namespace).Get(b.source.Name, metav1.GetOptions{})
 	if err != nil {
+		if b.source.Optional != nil && *b.source.Optional {
+			// ignore error when optional
+			return nil
```

Comment:
@pmorie can confirm, but I *think* we should actually continue setting up the volume (so the mount is there with the expected file permissions), and just leave it empty. If the configmap appears while the pod is running, the next resync should populate the volume